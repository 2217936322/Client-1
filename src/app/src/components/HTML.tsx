import React from 'react';
import { TMemo } from '@shared/components/TMemo';
import HTMLRender from 'react-native-render-html';
import { Dimensions } from 'react-native';

interface Props {
  html: string;
}
export const HTML: React.FC<Props> = TMemo((props) => {
  return (
    <HTMLRender
      html={props.html}
      imagesMaxWidth={Dimensions.get('window').width}
    />
  );
});
HTML.displayName = 'HTML';
