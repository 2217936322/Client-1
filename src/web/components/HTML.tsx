import React, { useMemo } from 'react';
import xss from 'xss';
import { Parser } from 'html-to-react';
import { TMemo } from '@shared/components/TMemo';

const parser = new Parser();

interface Props {
  html: string;
}
const HTML: React.FC<Props> = TMemo((props) => {
  const el = useMemo(() => {
    const html = xss.filterXSS(props.html, { css: false }); // 防止xss
    return parser.parse(html);
  }, [props.html]);

  return <div>{el}</div>;
});
HTML.displayName = 'HTML';

export default HTML;
