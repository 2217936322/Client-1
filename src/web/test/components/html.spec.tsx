import React from 'react';
import HTML from '@web/components/HTML';
import { mount } from 'enzyme';

describe('HTML Component', () => {
  test.each([
    ['<h1>Test Title</h1>'],
    ['<script src="http://example.com/script.js"></script>'],
    ['<style>*{color: red;}</style>'],
  ])('render "%s"', (originText) => {
    const wrapper = mount(<HTML html={originText} />);
    expect(wrapper.toJson()).toMatchSnapshot();
  });
});
