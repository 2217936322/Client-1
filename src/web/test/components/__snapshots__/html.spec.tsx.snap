// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Component render "<h1>Test Title</h1>" 1`] = `
<Memo(HTML)
  html="<h1>Test Title</h1>"
>
  <div>
    <h1
      key="0"
    >
      Test Title
    </h1>
  </div>
</Memo(HTML)>
`;

exports[`HTML Component render "<script src="http://example.com/script.js"></script>" 1`] = `
<Memo(HTML)
  html="<script src=\\"http://example.com/script.js\\"></script>"
>
  <div>
    &lt;script src="http://example.com/script.js"&gt;&lt;/script&gt;
  </div>
</Memo(HTML)>
`;

exports[`HTML Component render "<style>*{color: red;}</style>" 1`] = `
<Memo(HTML)
  html="<style>*{color: red;}</style>"
>
  <div>
    &lt;style&gt;*{color: red;}&lt;/style&gt;
  </div>
</Memo(HTML)>
`;
